src/App.js

```diff
@@ -46,6 +47,7 @@ class App extends Component {
           <Route exact path='/full-article' component={requireSignIn(FullArticle)}></Route>
           <Route exact path='/login' component={LoginForm}></Route>
           <Route exact path='/signup' component={SignUpForm}></Route>
+          <Route exact path='/user-page' component={UserPage}></Route>
```

Comment:
maybe you want to `requireSignIn` for this route

Reply:
Correct. Done.